### PR TITLE
bug_1256719 - Impl task to set release as readonly

### PR DESCRIPTION
**Pull request by allan-silva**
Depends on mozilla/balrog/pull/983
Depends new balrogclient release.

cc/ @mozbhearsum

Tested against local Balrog instance with following `task.json` sample:

```
{
    "scopes": ["project:releng:balrog:server:release", "project:releng:balrog:action:set-readonly"],
    "payload": {
        "product": "firefox",
        "version": "70.0.99",
        "build_number": 2
    }
}
```

### CODE REVIEW

**Comment by bhearsum** [comment]
Looks like we'll need a new `balrogclient` release before tests will pass. I'll take care of that this week.

**Comment by bhearsum** [comment]
When this is ready, @MihaiTabara is probably a good reviewer. It looks sane to me so far, though.

**Comment by bhearsum** [comment]
`balrogclient` release is done: https://pypi.org/project/balrogclient/1.0.0/

**Comment by allan-silva** [comment]
@MihaiTabara , @mozbhearsum this is ready for review.

File: balrogscript/requirements/base.in [hunk 1]
```diff
@@ -1,4 +1,4 @@
-balrogclient>=0.2.0
+balrogclient>=1.0.0
```
Comment (bhearsum) [inline balrogscript/requirements/base.in]: Let's bump this to 1.1.0 -- 1.0.0 still enforced its own dependency versions, which can cause some trouble when we deploy.

File: balrogscript/balrogscript/script.py [hunk 2]
```diff
@@ -187,6 +188,26 @@ def submit_toplevel(task, config, auth0_secrets):
...
+def create_state_updater(**kwargs):
+    return ReleaseStateUpdater(**kwargs)
+
+
+def set_readonly(task, config, auth0_secrets):
+    state_updater = create_state_updater(
+        api_root=config['api_root'],
+        auth0_secrets=auth0_secrets,
+        dummy=config['dummy'],
+        suffix=task['payload'].get('blob_suffix', ''),
```
Comment (bhearsum) [inline balrogscript/balrogscript/script.py]: Is this actually a feature we use, @MihaiTabara? If we don't, there's probably no reason to cargo cult it to this new run mode.

**Review by bhearsum — CHANGES_REQUESTED** [review]
Looks sane to me, let's bump the balrogclient version more though. And I want @MihaiTabara to review this, too.

File: balrogscript/balrogscript/script.py (hunk 2 above)
Comment (MihaiTabara) [reply balrogscript/balrogscript/script.py]: We don't use this anymore AFAIK, nor I see any occurrence of it in-tree in the taskgraph.

```
mtabara@mozspace:[]~/work/mozilla/clones/hg/fxtree/taskcluster$ grep -R 'blob_suffix' * 
mtabara@mozspace:[]~/work/mozilla/clones/hg/fxtree/taskcluster$ 
```

Let's remove it indeed.

File: balrogscript/requirements/base.in (hunk 1 above)
Comment (MihaiTabara) [reply balrogscript/requirements/base.in]: +1

**Review by MihaiTabara — APPROVED** [review]

**Comment by MihaiTabara** [comment]
Let's bump that `balrogclient` to `1.1.0` and remove the `blob_suffix` before we merge.

**Review by bhearsum — APPROVED** [review]

**Comment by MihaiTabara** [comment]
Merged this, thank you again for your contribution! It will be picked up by one of the deployments today or tomorrow as I'm planning on merging some other stuff too.

Will confirm with a message here.

**Pull request merged**